File: Android App - Shredsheets 2/app/src/main/java/revert/shredsheets/models/scales/Blues.java
package revert.shredsheets.models.scales;

public class Blues extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{ 3, 2, 1, 1, 3, 2};
    }

    @Override
    public String getName() {
        return "Blues";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{ "Minor Blues", "Major Blues" };
    }

    @Override
    protected boolean[][] getModalHighlighting() {
        return new boolean[][]{
                {true, true, false, true, true, true},
                {true, false, true, true, false, true},
        };
    }

    @Override
    public int[][] getDegrees() {
        return new int[][] {
                { 1,3,4,5,5,7 },
                { 1,2,3,3,5,6 }
        };
    }

    @Override
    public String[] getChords() {
        return new String[]{"m7", "M", "5 add ♭7", "m7", "M7", "m7"};
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/Blues.kt
package com.bm.shredsheets.models.scales

class Blues : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(3, 2, 1, 1, 3, 2)

    override val name: String
        get() = "Blues"

    override fun getModeNames(): Array<String> {
        return arrayOf("Minor Blues", "Major Blues")
    }

    override val getEnum: Scales
        get() = Scales.Blues

    override val modalHighlighting: Array<BooleanArray>
        protected get() = arrayOf(
            booleanArrayOf(true, true, false, true, true, true),
            booleanArrayOf(true, false, true, true, false, true))

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 3, 4, 5, 5, 7),
            intArrayOf(1, 2, 3, 3, 5, 6))

    override val chords: Array<String>
        get() = arrayOf("m7", "M", "5 add ♭7", "m7", "M7", "m7")
}